devices/ikea.js

```diff
@@ -49,6 +53,161 @@ const tradfriExtend = {
     }),
 };
 
+const manufacturerOptions = {manufacturerCode: herdsman.Zcl.ManufacturerCode.IKEA_OF_SWEDEN};
+
+const ikea = {
+    fz: {
+        air_purifier: {
+            cluster: 'manuSpecificIkeaAirPurifier',
+            type: ['attributeReport', 'readResponse'],
+            options: [exposes.options.precision('pm25'), exposes.options.calibration('pm25')],
+            convert: (model, msg, publish, options, meta) => {
+                const state = {};
+
+                if (msg.data.hasOwnProperty('particulateMatter25Measurement')) {
+                    const pm25Property = postfixWithEndpointName('pm25', msg, model);
+                    let pm25 = parseFloat(msg.data['particulateMatter25Measurement']);
+
+                    // Air Quality
+                    // Scale based on EU AQI (https://www.eea.europa.eu/themes/air/air-quality-index)
+                    // Using German IAQ labels to match the Develco Air Quality Sensor
+                    let airQuality;
+                    const airQualityProperty = postfixWithEndpointName('air_quality', msg, model);
+                    if (pm25 <= 10) {
+                        airQuality = 'excellent';
+                    } else if (pm25 <= 20) {
+                        airQuality = 'good';
+                    } else if (pm25 <= 25) {
+                        airQuality = 'moderate';
+                    } else if (pm25 <= 50) {
+                        airQuality = 'poor';
+                    } else if (pm25 <= 75) {
+                        airQuality = 'unhealthy';
+                    } else if (pm25 <= 800) {
+                        airQuality = 'hazardous';
+                    } else if (pm25 < 65535) {
+                        airQuality = 'out_of_range';
+                    } else {
+                        airQuality = 'unknown';
+                    }
```

Comment:
Not all of the above are in the `air_quality` expose, please add them.

Reply:
Good catch! I had a failing I forgot something but couldn't figure out what.